odxtools/singleecujob.py

```diff
@@ -142,9 +158,15 @@ def from_et(et_element, doc_frags: List[OdxDocFragment]) -> "NegOutputParam":
             dop_base_ref=dop_base_ref,
         )
 
-    def _resolve_references(self, odxlinks: OdxLinkDatabase) -> None:
+    def _build_odxlinks(self) -> Dict[OdxLinkId, Any]:
+        return {}
+
+    def _resolve_odxlinks(self, odxlinks: OdxLinkDatabase) -> None:
         self._dop = odxlinks.resolve(self.dop_base_ref)
 
+    def _resolve_snrefs(self, diag_layer: "DiagLayer") -> None:
+        pass
```

Comment:
Duplicate with lines 66-73

Reply:
for (probably) historic reasons, this file contains several classes. Line 66 is `InputParam`, this here is for `NegOutputParam`. This should be fixed in a future PR. (all files should be made to contain at most a single class; I also get annoyed by this quite frequently.)

Reply:
so maybe those lines need to go to a parent Param class?

Reply:
that might be a possibility for a future PR. I'm not really sure if this a terribly good idea though, because it makes things quite a bit less explicit, so I'd like to have a separate PR to assess that...